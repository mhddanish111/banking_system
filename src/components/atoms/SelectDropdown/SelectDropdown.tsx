import React from 'react';
import { isEmpty } from 'lodash';
import './SelectDropdown.scss';

type OptionProps = {
  label: string;
  value: string;
};

type SelectDropdownProps = {
  id: string;
  label: string;
  name: string;
  className?: string;
  value?: string;
  ariaLabel?: string;
  placeholder?: string;
  options: Array<OptionProps>;
  children?: React.ReactNode;
  onChange?: any;
  others?: any;
};

const SelectDropdown: React.FC<SelectDropdownProps> = (
  props: SelectDropdownProps,
) => {
  const {
    id,
    label,
    name,
    className,
    value,
    ariaLabel,
    children,
    placeholder,
    options,
    others,
    onChange,
  } = props;
  const getOption = () => {
    return (
      <>
        {placeholder && <option value="">{placeholder}</option>}
        {options.map((opt) => (
          <option key={opt.value} value={opt.value} aria-label={opt.label}>
            {opt.label}
          </option>
        ))}
      </>
    );
  };
  return (
    <>
      {label && <label htmlFor={id}>{label}</label>}
      <select
        id={id}
        name={name}
        className={`select-dropdown ${className}`}
        value={value}
        aria-label={ariaLabel}
        onChange={onChange}
        {...others}
      >
        {!isEmpty(children) && children}
        {isEmpty(children) && getOption()}
      </select>
    </>
  );
};

export default SelectDropdown;
