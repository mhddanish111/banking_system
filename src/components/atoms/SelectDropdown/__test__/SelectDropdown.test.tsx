import { shallow } from 'enzyme';
import SelectDropdown from '../SelectDropdown';

describe('<SelectDropdown />', () => {
  const onChange = jest.fn();
  const options = [{ label: 'test', value: 'test' }];
  const props = {
    className: '',
    value: 'test',
    ariaLabel: 'select dropdown',
    name: 'test',
    label: 'test',
    id: 'test',
    placeholder: 'Select',
    options,
    onChange,
  };
  const wrapper = shallow(<SelectDropdown {...props} />);
  it('should render the SelectDropdown Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
