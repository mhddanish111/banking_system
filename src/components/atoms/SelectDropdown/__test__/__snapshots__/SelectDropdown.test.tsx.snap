// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectDropdown /> should render the SelectDropdown Component correctly 1`] = `
<Fragment>
  <label
    htmlFor="test"
  >
    test
  </label>
  <select
    aria-label="select dropdown"
    className="select-dropdown "
    id="test"
    name="test"
    onChange={[MockFunction]}
    value="test"
  >
    <option
      value=""
    >
      Select
    </option>
    <option
      aria-label="test"
      key="test"
      value="test"
    >
      test
    </option>
  </select>
</Fragment>
`;
